src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -110,14 +114,67 @@ private IntPtr AllocException(Exception ex)
         [DllImport(JitSupportLibrary)]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
-        public static void Startup()
+
+        public static void Startup(TargetOS targetOS, TargetArchitecture targetArchitecture, string jitPathOverride)
+        {
+            Debug.Assert(s_jitLibrary == IntPtr.Zero);
+
+            if (!string.IsNullOrEmpty(jitPathOverride))
+            {
+                s_jitLibrary = NativeLibrary.Load(jitPathOverride);
+            }
+            else
+            {
+                string jitLibraryName = JitLibrary + '-' + GetTargetSpec(targetOS, targetArchitecture);
+                s_jitLibrary = NativeLibrary.Load(jitLibraryName, typeof(CorInfoImpl).Assembly, searchPath: null);
+            }
+
+            s_jitStartup = Marshal.GetDelegateForFunctionPointer<JitStartupDelegate>(NativeLibrary.GetExport(s_jitLibrary, "jitStartup"));
+            s_getJit = Marshal.GetDelegateForFunctionPointer<GetJitDelegate>(NativeLibrary.GetExport(s_jitLibrary, "getJit"));
+
+            s_jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+        }
+
+        private static string GetLibraryPrefix()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "" : "lib";
+        }
+
+        private static string GetLibraryExtension()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".dll";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return ".so";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return ".dylib";
+            }
+            else
+            {
+                throw new NotImplementedException("GetLibraryExtension");
+            }
+        }
+
+        private static string GetTargetSpec(TargetOS targetOS, TargetArchitecture targetArchitecture)
         {
-            jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+            string targetOSComponent = (targetOS == TargetOS.Windows ? "win" : "unix");
+            string targetArchComponent = targetArchitecture switch
+            {
+                TargetArchitecture.X64 => "x64",
+                TargetArchitecture.ARM64 => "arm64",
```

Comment:
Nit: I think you can fill all these in, even though we do not have them today, so that we do not have do it piecemeal later.

Reply:
Added in 22nd commit.